### PR TITLE
feat: Adding Skeleton component to component-library

### LINKED ISSUE #20874: Create component: Skeleton
### **Description**

Create `Skeleton` component based on audit and insight report 

### **Technical Details**
- TBC

### **Acceptance Criteria**
- Component exists built to specifications outlined in the insight report
 
### **References**
- TBC

### PR DESCRIPTION
**Pull request by georgewrmarshall**
Replacement of https://github.com/MetaMask/metamask-extension/pull/22278

## **Description**
Initial draft of the `Skeleton` component to review with the @MetaMask/design-system-engineers based on the findings in the [audit and component discovery sync](https://www.figma.com/file/bA81b0VvoGRgNS4E6059Pa/Loaders%2FSkeleton-Audit-and-Insight-Report?type=whiteboard&node-id=117%3A26&t=AnzMgSbVjzePPMmA-1)

## **Related issues**

Fixes: #20874 

## **Manual testing steps**

1. Go to the latest storybook build on this PR
2. Search `Skeleton` 
3. Check the docs, stories and controls

### **Before**

N/A

### **After**

https://github.com/user-attachments/assets/19e46ae0-d0ae-440a-9da4-99b037b6686b

Colors have been adjusted see [this comment](https://github.com/MetaMask/metamask-extension/pull/29082/files#r1882837341)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/component-library/skeleton/skeleton.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+      backgroundColor={BackgroundColor.backgroundAlternative}
+      borderRadius={BorderRadius.LG}
+      padding={4}
+    >
+      <Skeleton height={32} width="100%" />
+      <Skeleton height={16} width="95%" />
+      <Skeleton height={16} width="95%" />
+    </Skeleton>
+  ),
+};
```
Comment (georgewrmarshall) [inline ui/components/component-library/skeleton/skeleton.stories.tsx:71]: This story should be fixed when this PR gets merged https://github.com/MetaMask/metamask-extension/pull/27783

**Comment by Akatori-Design** [comment]
@georgewrmarshall If its just pulsing animation then easing can be `linear` or u might not have to specify at all?

Also, how is it lighter grey in dark mode if its using `background/alternative`?

File: ui/components/component-library/skeleton/skeleton.scss [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  animation: skeleton-pulse 2s cubic-bezier(0.4, 0, 0.6, 1) infinite;
+  background-clip: padding-box;
+  cursor: default;
+  pointer-events: none;
+  user-select: none;
+
+  &--hide-children * {
+    visibility: hidden;
+  }
+}
```
Comment (brianacnguyen) [inline ui/components/component-library/skeleton/skeleton.scss:22]: were these cleared with @Akatori-Design ?

**Review by brianacnguyen — CHANGES_REQUESTED** [review]
1 small question

**Comment by georgewrmarshall** [comment]
> @georgewrmarshall If its just pulsing animation then easing can be `linear` or u might not have to specify at all?
> 
> Also, how is it lighter grey in dark mode if its using `background/alternative`?

Good question @Akatori-Design it's using iconMuted but pulsing between 20% and 50% opacity. Should it be using background alternative?

https://github.com/MetaMask/metamask-extension/pull/29082/files#diff-dce96c1096d70fd28631f164a82f57a16e43470e1a7a2fffbc08510f2e5be2c8R33

File: ui/components/component-library/skeleton/skeleton.scss (hunk 2 above)
Comment (georgewrmarshall) [reply ui/components/component-library/skeleton/skeleton.scss:22]: Getting him to review also

Comment (georgewrmarshall) [reply ui/components/component-library/skeleton/skeleton.scss:22]: Good call I'm asking @Akatori-Design to review as well https://github.com/MetaMask/metamask-extension/pull/29082#issuecomment-2537265842

Comment (georgewrmarshall) [reply ui/components/component-library/skeleton/skeleton.scss:22]: Updated after syncing with @Akatori-Design on easing, timing and opacity.


https://github.com/user-attachments/assets/3528cd88-f18f-4238-ad1b-d1490cab1e93

**Comment by Akatori-Design** [comment]
✅ **DESIGN APPROVED**

color : `icon.alternative` (20% <--> 10%)
duration : `1400ms`
ease : `linear`(`cubic-bezier(0, 0, 1, 1);`)

All above updates confirmed in sync with @georgewrmarshall 

Looks good to me!

**Review by brianacnguyen — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**